clang/lib/CConv/Constraints.cpp

```diff
@@ -35,8 +35,25 @@ Constraint::Constraint(ConstraintKind K, std::string &Rsn,
 
 // Remove the constraint from the global constraint set.
 bool Constraints::removeConstraint(Constraint *C) {
+  bool RetVal = false;
   removeReasonBasedConstraint(C);
-  return constraints.erase(C) != 0;
+  RetVal = constraints.erase(C) != 0;
+  // Delete from graph.
+  ConstraintsGraph *TG = nullptr;
+  if (Geq *GE = dyn_cast<Geq>(C)) {
+    assert(GE != nullptr && "Invalid constrains requested to be removed.");
+    // We can only remove constraints from ConstAtoms.
+    if (GE != nullptr && isa<ConstAtom>(GE->getRHS()) &&
+        isa<VarAtom>(GE->getLHS())) {
+      TG = GE->constraintIsChecked() ? ChkCG : PtrTypCG;
+      ConstraintsGraph::vertex_t SrcIdx = TG->addVertex(GE->getRHS());
+      ConstraintsGraph::vertex_t DstIdx = TG->addVertex(GE->getLHS());
+      RetVal = true;
+      // Remove the edge form the corresponding constraint graph.
+      boost::remove_edge(SrcIdx, DstIdx, TG->CG);
+    }
+  }
```

Comment:
Should it be an assertion failure if you are trying to remove a non `Geq` constraint?

Reply:
As of now, yes. We don't know the reason why we would like to remove other types of constraints.

Reply:
So let's add the assert? I don't see one there. That's why I was asking.

Reply:
The assert is added in the caller. The `ConstraintsGraph` does not know about `Geq` constraints, so the check should be done in the caller (`Constraints::removeConstraint(Constraint *C)`) and is done.